src/whitelist.ts

```diff
@@ -0,0 +1,35 @@
+import axios from 'axios';
+import { config } from './config.js';
+
+let whitelistCache: Set<string> = new Set();
+let lastFetchTime = 0;
+
+export async function fetchWhitelist() {
+  if (!config.whitelist.enabled) {
+    whitelistCache.clear();
+    return whitelistCache;
+  }
+
+  const now = Date.now();
+  if (now - lastFetchTime < config.whitelist.fetchDelay * 1000) {
+    return whitelistCache;
+  }
+
+  try {
+    const response = await axios.get(`https://${config.whitelist.domain}/.well-known/nostr.json`);
+    const data = response.data;
+    if (data && data.names) {
+      whitelistCache = new Set(Object.values(data.names));
+      lastFetchTime = now;
+    }
+  } catch (error) {
+    console.error("Failed to fetch whitelist:", error);
+  }
+
+  return whitelistCache;
+}
+
+export function isWhitelisted(pubkey: string): boolean {
```

Comment:
I would convert this method to `async` and make it dynamically fetch / refresh the whitelist when needed. that way the API endpoints themselves do not need to think about getting the whitelist